src/main/java/randoop/generation/ComponentManager.java

```diff
@@ -222,7 +258,15 @@ SimpleList<Sequence> getSequencesForType(TypedOperation operation, int i, boolea
         }
       }
 
-      if (packageLiterals != null) {
+      // If the input selector in {@link ForwardGenerator} is GRT Constant Mining and the p_const
+      // coin flip succeeds, set literals to only the component sequences that are class-level
+      // extracted literals from the declaring class. That is, don't add literals from the package
+      // level.
+      boolean shouldOnlyIncludeConstantsFromDeclaringClass =
+          GenInputsAbstract.input_selection == GenInputsAbstract.InputSelectionMode.CONSTANT_MINING
+              && Randomness.weightedCoinFlip(GenInputsAbstract.p_const);
+
```

Comment:
I'm a bit confused on this implementation: from the GRT paper, it sounds like there are 2 levels of constant usage. The global usage constants are always considered. For the local constants, they are each added to the pool for a MUT with probability $p_const$. 

It seems like this implementation will exclude the global level constants with probability $p_const$? This seems like the inverse of design in the GRT paper, but perhaps I am misunderstanding something